src/main/java/com/uber/profiling/YamlConfigProvider.java

```diff
@@ -172,6 +171,20 @@ private static void addConfig(Map<String, Map<String, List<String>>> config, Str
             configValueList.add(value.toString());
```

Comment:
Could you check value instance of Map here, and call addMapConfig? So we keep addConfig as the single point to add config.